app/workers/send_email_worker.rb

```diff
@@ -4,13 +4,21 @@ class SendEmailWorker < ApplicationWorker
   RATE_LIMIT_THRESHOLD = 21_600 # max requests in a minute, equates to 350 a second
   RATE_LIMIT_INTERVAL = 60
 
-  sidekiq_options retry: 9
-
   sidekiq_retries_exhausted do |msg|
     Email.find_by(id: msg["args"].first, status: :pending)
          &.update!(status: :failed)
   end
 
+  sidekiq_options lock: :until_executing,
+                  unique_across_queues: true,
+                  unique_args: :uniqueness_with, # in upcoming version 7 of sidekiq-unique-jobs, :unique_args is replaced with :lock_args
+                  on_conflict: :log,
+                  retry_count: 9
```

Comment:
I was thinking that we'd probably want something to automatically release the lock after a certain amount of time - just to really try cut out the possibility that a lock is held without the job being in Sidekiq which would block the future attempt from recovering.

However looking through version 6 of sidekiq-unique-jobs that [doesn't look to be possible](https://github.com/mhenrixon/sidekiq-unique-jobs/blob/4ce41b33f53662671d34c3ccc44aee7642e170bc/redis/lock.lua#L50-L60). Perhaps it is in [version 7](https://github.com/mhenrixon/sidekiq-unique-jobs/blob/master/lib/sidekiq_unique_jobs/lua/lock.lua#L74-L89) - we may want to document that there is an unlikely possibility that jobs can get perpetually stuck.

All looks good besides that

Reply:
Thanks for the comment Kev!

Hmm your suggestion does make sense and it's a shame it's not available in the current version - I've added a mention in the commit around the concern.